src/sentry/tasks/collect_project_platforms.py

```diff
@@ -12,16 +12,17 @@
 def collect_project_platforms(**kwargs):
     now = timezone.now()
 
-    min_project_id = 0
-    max_project_id = Project.objects.using_replica().aggregate(x=Max("id"))["x"] or 0
-    step = 1000
-    while min_project_id <= max_project_id:
+    all_project_query = [p.id for p in Project.objects.using_replica().all()]
```

Comment:
@RyanSkonnord Django's paginator doesn't seem to like BaseQuerySets so would it make sense to get a list of all project ids first?

Reply:
No, that's too much to load into memory.

I could follow up about Django's problem with BaseQuerySets, but the pagination logic should be simple enough to hack together something small. See https://docs.djangoproject.com/en/dev/topics/db/queries/#limiting-querysets for `LIMIT` clauses on Django queries. I think this will work:

```python
def paginate_projects_by_id():
    id_cursor = -1
    while True:
        queryset = Project.objects.filter(id__gt=id_cursor).order_by("id")
        page = list(queryset[:1000])
        if not page:
            return
        yield page
        id_cursor = page[-1].id
```

A slightly simpler solution, but possibly less performant:

```python
def paginate_projects_by_index():
    index_cursor = 0
    while True:
        queryset = Project.objects.all().order_by("id")
        page = list(queryset[index_cursor : index_cursor + 1000])
        if not page:
            return
        yield page
        index_cursor += 1000
```